### PR TITLE
Ensure doses tracked at pod end of life

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-966

### CODE REVIEW

**Comment by ps2** [comment]
Dose tracking fixes to ensure total delivery calculation is accurate.

**Review by rickpasetto — COMMENTED** [review]
Hm.  This doesn't seem to be in the area of code we had chatted about, @ps2 , so I'm not sure I can approve without more of an explanation.  Would it make sense to talk live?

**Comment by ps2** [comment]
See the LoopKit changes for what we talked about; this is a somewhat tangential issue that was impacting the calculation that I discovered while working on the LoopKit changes. I can create a separate ticket for these changes, if you feel that's necessary.

**Comment by rickpasetto** [comment]
@ps2 ah, no, no need for another ticket, that explains it.  I'll take another look at this one.

File: DashKit/DashPumpManagerState.swift [hunk 1]
```diff
@@ -213,6 +193,19 @@ public struct DashPumpManagerState: RawRepresentable, Equatable {
         podActivatedAt = status.expirationDate - .days(3)
         podTotalDelivery = status.delivered
     }
+    
+    mutating func finalizeDoses() {
+        if let bolus = unfinalizedBolus, bolus.isFinished(at: dateGenerator()) {
```
Comment (rickpasetto) [inline DashKit/DashPumpManagerState.swift:198]: Is there a unit test that checks both of these (or rather all 4 of these) code paths?

**Review by rickpasetto — APPROVED** [review]
Looks okay to me, but I admit I'm not 100% sure of correctness, because of my lack of domain knowledge here.  Do the unit tests cover everything, including the new code?

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**